sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/csharp/CSharpGenerator.java

```diff
@@ -1540,4 +1553,309 @@ private String generateLiteral(final PrimitiveType type, final String value)
+            Separators.FIELD.appendToGeneratedBuilder(sb, indent, "builder");
+
+            i += varDataToken.componentTokenCount();
+        }
+
+        if (lengthBeforeLastGeneratedSeparator > -1)
+        {
+            sb.setLength(lengthBeforeLastGeneratedSeparator);
+        }
+    }
+
+    private int writeTokenDisplay(
+        final String fieldName,
+        final Token typeToken,
+        final StringBuilder sb,
+        final String indent)
+    {
+        if (typeToken.encodedLength() <= 0)
+        {
+            return -1;
+        }
+
+        append(sb, indent, "builder.Append(\"" + fieldName + Separators.KEY_VALUE + "\");");
+
+        if (typeToken.isConstantEncoding())
+        {
+            append(sb, indent, "builder.Append(\"" + typeToken.encoding().constValue() + "\");");
+        }
+        else
+        {
+            switch (typeToken.signal())
+            {
+                case ENCODING:
+                    if (typeToken.arrayLength() > 1)
+                    {
+                        if (typeToken.encoding().primitiveType() == PrimitiveType.CHAR)
+                        {
+                            append(sb, indent, "for (int i = 0; i < " + fieldName +
+                                "Length && this.Get" + fieldName + "(i) > 0; ++i)");
+                            append(sb, indent, "{");
+                            append(sb, indent, "    builder.Append((char)this.Get" + fieldName + "(i));");
+                            append(sb, indent, "}");
+                        }
+                        else
+                        {
+                            Separators.BEGIN_ARRAY.appendToGeneratedBuilder(sb, indent, "builder");
+                            append(sb, indent, "for (int i = 0; i < " + fieldName + "Length; ++i)");
+                            append(sb, indent, "{");
+                            append(sb, indent, "    if (i > 0) builder.Append(',');");
+                            append(sb, indent, "    builder.Append(Get" + fieldName + "(i));");
+                            append(sb, indent, "}");
+                            Separators.END_ARRAY.appendToGeneratedBuilder(sb, indent, "builder");
+                        }
+                    }
+                    else
+                    {
+                        append(sb, indent, "builder.Append(this." + fieldName + ");");
+                    }
+                    break;
+
+                case BEGIN_ENUM:
+                    append(sb, indent, "builder.Append(this." + fieldName + ");");
+                    break;
+
+                case BEGIN_SET:
+                    append(sb, indent, "this." + fieldName + ".BuildString(builder);");
+                    break;
+
+                case BEGIN_COMPOSITE:
+                {
+                    append(sb, indent, "if (this." + fieldName + " != null)");
+                    append(sb, indent, "     this." + fieldName + ".BuildString(builder);");
```

Comment:
Blocks without braces is not good practice and easy to introduce bugs with.

Reply:
Done. I also saw an small indenting issue which I fixed.